Disable scm-sync plugin and enabled jcasc

alvagante opened the pull request:
Jcasc plugin configurations missing.

alvagante committed:
Removed redundat jenkins class and disabled scm-sync plugin

alvagante committed:
Add Jenkins casc class to jenkins role

alvagante commented:
Tested only locally, configuration can be managed with params like:
psick::jenkins::jcasc::config_template: 'profile/jenkins/jcasc/jenkins.yaml.erb'
psick::jenkins::jcasc::options_hash:
  privateKey: 'crypted_data'
  url : 'jenkins.sandbox.graduway.com'
  ...

antgel commented on data/role/jenkins-dev.yaml:
```diff
@@ -1,10 +1,7 @@
 
 # Enabled and configure scm-sync plugin
-psick::jenkins::scm_sync_repository_url: git@github.com:graduway/jenkins-config-dev.git
+DISABLEpsick::jenkins::scm_sync_repository_url: git@github.com:graduway/jenkins-config-dev.git
```
@alvagante Is `DISABLE` valid syntax? Is there a reason why we don't just delete the line completely?

alvagante replied:
@antgel it's a quick way to "comment" the entry, could be any string, or also a normal # .
Anyway can be totally removed, but there are other hiera settings which might be redundant, and actually, now that we have our data separated, we can clean up a lot of sample entries in hieradata module, if you want

antgel replied:
@alvagante Thanks. Please edit the commit to remove the unused code instead of "commenting" it out, this is a general principle of our workflow. I'll create an Issue for the further cleanup, it's outside scope of this PR.

olegy-ait replied:
for prod, qa, sandbox leave it as is. Change it when we configure jcasc for these envs.
For dev removed at all.

antgel reviewed (changes requested):
More of a question than a rejection...

alvagante committed:
Enable PuppetDB auto ssl-setup

alvagante commented on data/role/puppet_foss_master.yaml:
```diff
@@ -6,4 +6,4 @@ psick::profiles::linux_classes:
 psick::puppet::autosign::autosign: 'on'
 psick::puppet::autosign::autosign_match: "*.%{::env_region}.graduway.com"
-
+psick::puppet::foss_master::enable_puppetdb_sslsetup: true
```
This is the configuration option that enables the execution of puppetdb ssl-setup , it's valid with current upstream psick module.

Note that at the moment the complete setup of puppetdb might require more than a single puppet run, and the first puppet run could have errors on puppetdb service

alvagante committed:
Cleanup unused jenkins data

To verify if ssh keys created for user jenkins with psick::jenkins::ssh_private_key_content
and psick::jenkins::ssh_public_key_content are necessary or were used only for the old scm-sync plugin.

olegy-ait committed:
Add Jenkins configuration as code template

This commit adds jenkins.yaml.erb template for
Jenkins Configuration as Code Plugin
(https://github.com/jenkinsci/configuration-as-code-plugin)
Also this commit configures jenkins dev to use jcasc plugin with
the mentioned template.
For other envs it reverts back SCM plugin usage.
We will change these envs when DEV will be OK.

olegy-ait committed:
Do not install openssh by PSICK on windows slave

This commit removes openssh installation by PSICK because PSICK
tries to install unix openssh but jnkins slave is windows.

alvagante commented on data/role/jenkins.yaml:
```diff
@@ -1,17 +1,20 @@
   ec2_tags:
     template: 'profile/aws.tags.erb'
+
+psick::jenkins::jcasc::config_template: 'hieradata/jenkins.yaml.erb'
```
+1 for having the template directly in the hieradata module.
Just note that in other cases templates were placed in the profile module in the control-repo (https://github.com/graduway/psick/tree/production/site/profile/templates) so it makes sense to standardise the approach.

Also, I'd suggest to have coherent naming for files and dirs in the templates dir with the classes were they are used.
So this could become  :
psick::jenkins::jcasc::config_template: 'hieradata/jenkins/jenkins.yaml.erb'
and the template moved accordingly.
my2c

antgel replied:
Thanks @alvagante , consistent naming is indeed important.

@olegy-ait 

1. Please open an Issue to move the other templates into hieradata.
2. Please remove useless merge commit #did2c91, there is no reason to merge a branch into itself. If you need those changes, then the existing commits on the branch should be rebased / edited to include them.

olegy-ait commented:
Merge commit has been replaced to rebase.
Issue #101 has been created.

antgel commented:
@alvagante Can you please approve (assuming it's okay) the PR, then I will dismiss my reject and merge it.

alvagante commented:
@antgel Being the original PR author github prevents me from approving it. 
ANyway the jcasc settings look syntactically ok to me, so +1 for me
